### PR TITLE
[14.0] fix sale_stock_picking_blocking: partner_commercial_id field

### PR DESCRIPTION
**Pull request by damdam-s**

### CODE REVIEW

File: sale_stock_picking_blocking/models/sale_order.py [hunk 1]
```diff
@@ -30,7 +30,9 @@ def onchange_partner_id(self):
         """Add the 'Default Delivery Block Reason' if set in the partner."""
         res = super().onchange_partner_id()
         for so in self:
-            so.delivery_block_id = so.partner_id.default_delivery_block or False
+            so.delivery_block_id = (
+                so.commercial_partner_id.default_delivery_block or False
```
Comment (jcoux) [inline sale_stock_picking_blocking/models/sale_order.py]:
```suggestion
                so.partner_id.commercial_partner_id.default_delivery_block or False
```

File: sale_stock_picking_blocking/models/sale_order.py [hunk 2]
```diff
@@ -46,8 +48,11 @@ def action_remove_delivery_block(self):
     def copy(self, default=None):
         new_so = super().copy(default=default)
         for so in new_so:
-            if so.partner_id.default_delivery_block and not so.delivery_block_id:
-                so.delivery_block_id = so.partner_id.default_delivery_block
+            if (
+                so.commercial_partner_id.default_delivery_block
+                and not so.delivery_block_id
+            ):
+                so.delivery_block_id = so.commercial_partner_id.default_delivery_block
```
Comment (jcoux) [inline sale_stock_picking_blocking/models/sale_order.py]:
```suggestion
            if (
                so.partner_id.commercial_partner_id.default_delivery_block
                and not so.delivery_block_id
            ):
                so.delivery_block_id = so.partner_id.commercial_partner_id.default_delivery_block
```

**Review by jcoux — CHANGES_REQUESTED** [review]
You can't use directly commercial partner on sale order because this field provides from:
https://github.com/OCA/sale-workflow/tree/14.0/sale_commercial_partner
And you don't have (and I think don't want) a dependency on it.

**Comment by damdam-s** [comment]
thanks @jcoux 
nice catch.

Fixed now

**Review by jcoux — APPROVED** [review]
Thanks :+1:

**Review by rousseldenis — APPROVED** [review]
A little test should avoid regression

**Comment by rafaelbn** [comment]
Hello @damdam-s , do you want to add the test for line 55? thank you!

**Comment by rafaelbn** [comment]
> Hello @damdam-s , do you want to add the test for line 55? thank you!

@damdam-s do you want to addd test as @rousseldenis said 

> A little test should avoid regression

 or not ? Just quiestion to go ahead with this one! 😄 

Thank you!

**Comment by damdam-s** [comment]
@rafaelbn hello. I won't have much time to add those tests :( 
Feel free to add them and superseed this PR.

**Pull request merged**